### PR TITLE
Build more integration tests during pre-submit

### LINKED ISSUE #76419: Run flutter framework post-submit tests on PRs before they are merged
This has been a source of delays with the Engine->Framework roller this week. There are tests which were failing after being merged in, which weren't being run as part of the pre-submit tests.

### PR DESCRIPTION
**Pull request by jmagman**
Add the missing files that should be checked in to `dev/integration_tests/ui/macos` to make it build.  Also add `examples/image_list/android/gradle/wrapper/gradle-wrapper.properties` which should be checked in.

[Linux](https://ci.chromium.org/p/flutter/builders/try/Linux%20build_tests/11407?) went from 17 to 19 minutes, [Mac](https://ci.chromium.org/p/flutter/builders/try/Mac%20build_tests/8020?) went from 17 to 27, and [Windows](https://ci.chromium.org/p/flutter/builders/try/Windows%20build_tests/9044?) seems the same.  I'll add a [fourth subshard](https://github.com/flutter/infra/blob/bb5fafafa21daedef60bf9d9e80e67ec2f5be096/config/framework_config.star#L504) for Mac, based on those numbers.

Fixes https://github.com/flutter/flutter/issues/76419

### CODE REVIEW

File: dev/bots/test.dart [hunk 1]
```diff
@@ -345,8 +345,15 @@ Future<void> _runWebToolTests() async {
 Future<void> _runBuildTests() async {
   final List<FileSystemEntity> exampleDirectories = Directory(path.join(flutterRoot, 'examples')).listSync()
     ..add(Directory(path.join(flutterRoot, 'packages', 'integration_test', 'example')))
+    ..add(Directory(path.join(flutterRoot, 'dev', 'integration_tests', 'abstract_method_smoke_test')))
```
Comment (jonahwilliams) [inline dev/bots/test.dart:348]: I think this one (and possibly others) are running on firebase test lab.

**Review by jonahwilliams — APPROVED** [review]
LGTM (but I would look into the android situation)

File: dev/bots/test.dart (hunk 1 above)
Comment (jmagman) [reply dev/bots/test.dart:348]: I totally forgot those run on pre-submit, good call.

Comment (jmagman) [reply dev/bots/test.dart:348]: That's the only one.

**Pull request merged**